base/base_tests/string_utils_test.cpp

```diff
@@ -1276,3 +1276,28 @@ UNIT_TEST(Trim)
   std::string strWithLeftRightSpaces = "  " + kStrWithoutSpaces + "  ";
   TEST_EQUAL(strings::Trim(strWithLeftSpaces), kStrWithoutSpaces, ());
 }
+
+UNIT_TEST(GetSocialPageNickname)
+{
+  TEST_EQUAL(strings::GetSocialPageNickname("some_page"), "@some_page", ());
+  TEST_EQUAL(strings::GetSocialPageNickname("facebook.com/some_page"), "@some_page", ());
```

Comment:
The better idea is to draw read-only @ symbol to the left of the editable field, and do not confuse users how they should type it in the editor.